kong/db/schema/typedefs.lua

```diff
@@ -525,6 +525,20 @@ typedefs.headers = Schema.define {
 
 typedefs.no_headers = Schema.define(typedefs.headers { eq = null } )
 
+typedefs.version = Schema.define {
+  type = "string",
+  match_any = {
+    patterns = { "^%d+[%.%d]*$", "^%d+[%.%d]*%-?.*$", },
+    err = "invalid version number: must be in format of X.Y.Z",
+  },
+  match_none = {
+    {
+      pattern = "%.%.",
+      err = "must not have empty version segments"
+    },
+  },
+}
```

Comment:
1. semantic_version might be a better name for this field.
2. Please ensure that this works for Kong Enterprise. There is a literal after the patch number for Kong Enterprise.

Reply:
Name was changed. This does work with suffix, https://github.com/Kong/kong/pull/6612/files#diff-97419d4c3d96f2fce4c8bcf5d1222b6a8e55bb38efc91855b8d7c7311a8cd042R152 proves it.

Reply:
That's not the correct test case. Enterprise versions are of the form  `1.2.3.4-enterprise-edition`

Reply:
The test case has been fixed.